hangman.py

```diff
@@ -0,0 +1,338 @@
+    It returns true or false, which is understood as winning or losing in hangman.
+    '''
+    while True:
+        print("--------------------------------------")
+        if guesses_left <= 0:
+            return False
+        guessed_letter = input_information(guesses_left, letters_guessed)
+        if hints_switch and guessed_letter == SYMBOL_FOR_HINTS:
+            #Game with hints activated
+            show_possible_matches(get_guessed_word(secret_word, letters_guessed))
+            continue
+        if guessed_letter.isalpha():
+            guessed_letter = guessed_letter.lower()
+        else:
+            #System of losing warnings/guesses activated
+            if warnings_left == 0:
+                guesses_left = not_correct_input_losing_guesses(guesses_left, secret_word, letters_guessed)
+            else:
+                warnings_left = not_correct_input_losing_warnings(guesses_left, secret_word, letters_guessed)
+            continue
+        if guessed_letter in letters_guessed:
+            #System of losing warnings/guesses activated
+            if warnings_left == 0:
+                guesses_left = guessed_before_losing_guesses(guesses_left, letters_guessed)
+            else:
+                warnings_left = guessed_before_losing_warnings(warnings_left, letters_guessed)
+            continue
+        else:
+            letters_guessed.add(guessed_letter)
+            guesses_left = results(guessed_letter, guesses_left, letters_in_secret_word, secret_word, letters_guessed)
+        if is_word_guessed(secret_word, letters_guessed):
+            return True
+
+def hangman(secret_word, hints_switch = False):
+    '''
+    secret_word: string, the secret word to guess.
+
+    Starts up an interactive game of Hangman.
+
+    * At the start of the game, let the user know how many
+      letters the secret_word contains and how many guesses s/he starts with.
+
+    * The user should start with 6 guesses
+
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+
+    * Ask the user to supply one guess per round. Remember to make
+      sure that the user puts in a letter!
+
+    * The user should receive feedback immediately after each guess
+      about whether their guess appears in the computer's word.
+
+    * After each guess, you should display to the user the
+      partially guessed word so far.
+
+    Follows the other limitations detailed in the problem write-up.
+    '''
+    letters_in_secret_word = set(secret_word)
+    start_of_the_game()
+    guesses_left = GUESSES_AT_START
+    warnings_left = WARNINGS_AT_START
+    letters_guessed = set()
+    winning = main_match_win_or_lose(guesses_left, letters_guessed, warnings_left, letters_in_secret_word, hints_switch)
+    end_of_the_game(winning, guesses_left, secret_word)
+
+
+def match_with_gaps(my_word, other_word, my_word_for_dublicate):
```

Comment:
I believe you don't need this `my_word_for_dublicate`

Reply:
done